src/qofono.cpp

```diff
@@ -403,8 +1928,24 @@ QOfono::QOfono()
 
 QString QOfono::mobileCountryCodeToAlpha2CountryCode(int mcc)
 {
-    if (mcc < 0 || mcc >= COUNTRY_CODES_LENGTH) {
-        qWarning() << Q_FUNC_INFO << "given unknown Mobile Country Code:" << mcc;
+    int low = 0;
+    int high = (sizeof(qofonoMccList)/sizeof(qofonoMccList[0]))-1;
+
+    while (low <= high) {
+        int mid = (low + high)/2;
+        int val = qofonoMccList[mid].mcc;
+        if (val < mcc) {
+            low = mid + 1;
+        } else if (val > mcc) {
+            high = mid - 1;
+        } else {
+            /* MCC is found. Back up a litte to the first entry */
+            while (mid > 0 && qofonoMccList[mid-1].mcc == val) mid--;
+            return QString(qofonoMccList[mid].cc);
```

Comment:
No objecting, but was there a particular reason for backing up? Wouldn't any item do because they have the same MCC & alpha country string?